nornir_nautobot/plugins/tasks/dispatcher/default.py

```diff
@@ -792,13 +795,70 @@ def get_commands(
                         raise NornirNautobotException(error_msg)
                 command_results.update({command: result[0].result})
             except NornirSubTaskError as exc:
-                error_code = EXCEPTION_TO_ERROR_MAPPER.get(type(exc.result.exception), "E1016")
+                error_code = EXCEPTION_TO_ERROR_MAPPER.get(type(exc.result.exception), "E1014")
                 error_msg = get_error_message(error_code, exc=exc)
                 logger.error(error_msg, extra={"object": obj})
                 raise NornirNautobotException(error_msg)
 
         return Result(host=task.host, result={"output": command_results})
 
+    @classmethod
+    def get_command_with_prompts(
+        cls,
+        task: Task,
+        logger,
+        obj,
+        command,
+        prompts,
+        regex_flags=re.IGNORECASE,
+        escape_sequence=chr(3),  # Ctrl-C
+        **kwargs,
+    ):  # pylint: disable=too-many-positional-arguments, too-many-locals
+        """A task to run a command on a device and react to resulting prompts.
+
+        Args:
+            task (Task): Nornir Task.
+            logger (logging.Logger): Logger that may be a Nautobot Jobs or Python logger.
+            obj (Device): A Nautobot Device Django ORM object instance.
+            command (str): A command to execute.
+            prompts (dict): A dictionary of regex pattern prompts and responses.
+            regex_flags (re.RegexFlag): Flags to pass to re.search.
+            escape_sequence (str): The escape sequence to send if no prompt is matched.
+            kwargs (dict): Additional keyword arguments to pass to Netmiko.
+        """
+        logger.debug(f"Executing get_command_with_prompts for {task.host.name} on {task.host.platform}")
+
+        net_connect = task.host.get_connection("netmiko", task.nornir.config)
+        base_prompt = net_connect.find_prompt()
+        # We disable stripping the prompt to detect if the command is complete
+        last_output = net_connect.send_command_timing(command, strip_prompt=False, **kwargs)
```

Comment:
What command are we sending here? How is command different then prompt?

Reply:
The command is the original cli command that would trigger a prompt on the device (e.g., `copy scp: flash:`)
The prompt is a response from the device like "Enter password:" that requires a response to complete the original command